{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# APEX QTL association testing\n",
    "\n",
    "This notebook implements a workflow for using [APEX](https://www.biorxiv.org/content/10.1101/2020.12.18.423490v1) to conduct analysis. APEX implements a linear mixed model for association testing. This is potentially useful for analysis with related individuals.\n",
    "\n",
    "**In our pilot cis-eQTL analysis with around ~500 samples we found that APEX is less robust compared to tensorQTL for association scans and for linear regression using OLS there is no speed advantage over tensorQTL. Gene level p-value from APEX are on average smaller than that from tensorQTL, resulting in more genes discovered. However, comparison with univariate fine-mapping seem to suggest that these p-values are a bit inflated. We therefore recommend using tensorQTL for at least the analysis of samples without related individuals.**\n",
    "\n",
    "## Other cautions\n",
    "\n",
    "- `--low-mem` option does not work for APEX so we do not use it (https://github.com/corbinq/apex/issues/7).\n",
    "- Also notice that the command options are different from those on the APEX website documentation. The commands on the documentation page does not work (last updated September 2021). The commands below were constructed and tested by our team based on our understanding of the program, without input from APEX authors."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Input\n",
    "\n",
    "- List of molecular phenotype files: a list of `bed.gz` files containing the table describing the molecular phenotype. It should have a companion index file in `tbi` format.\n",
    "- List of genotypes in VCF format for each chromosome, those vcf are converted beforehand from plink trio in the data_processing sections\n",
    "- List of GRM containing path to GRM matrices that generated by the [GRM module](https://cumc.github.io/xqtl-pipeline/code/data_preprocessing/genotype/GRM.html)\n",
    "- Covariate file, a file with #id + samples name as colnames and each row a covariate: fixed and known covariates as well as hidden covariates recovered from factor analysis.\n",
    "\n",
    "## Output\n",
    "\n",
    "For each chromosome, several of summary statistics files are generated, including both nomial test statistics for each test, as well as region (gene) level association evidence.\n",
    "\n",
    "The columns of nominal association result are as follows:\n",
    "\n",
    "- #chrom : Variant chromosome.\n",
    "- pos : Variant chromosomal position (basepairs).\n",
    "- ref : Variant reference allele (A, C, T, or G).\n",
    "- alt : Variant alternate allele.\n",
    "- gene : Molecular trait identifier (as specified in --bed {trait-file}).\n",
    "- beta : OLS regression slope for variant on trait.\n",
    "- se : Standard error of regression slope.\n",
    "- pval : Single-variant association nominal p-value.\n",
    "- variant_id: ID of the variant (rsid or chr:position:ref:alt)\n",
    "\n",
    "The columns of region (gene) level association evidence are as follows:\n",
    "\n",
    "- #chrom : Molecular trait chromosome.\n",
    "- start : Molecular trait start position.\n",
    "- end : Molecular trait end position.\n",
    "- gene : Molecular trait identifier.\n",
    "- gene_pval : Trait-level p-value calculated across all variants in the cis region using the Cauchy combination test, comparable to beta-approximated permutation p-values.\n",
    "- n_samples : Number of samples included in analysis.\n",
    "- n_covar : Number of covariates included in analysis, including intercept.\n",
    "- resid_sd : Square root of regression mean squared error under the null model.\n",
    "- n_cis_variants : Number of variants in the cis region (which were used to calculate gene_pval)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Command interface"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: sos run APEX.ipynb [workflow_name | -t targets] [options] [workflow_options]\n",
      "  workflow_name:        Single or combined workflows defined in this script\n",
      "  targets:              One or more targets to generate\n",
      "  options:              Single-hyphen sos parameters (see \"sos run -h\" for details)\n",
      "  workflow_options:     Double-hyphen workflow-specific parameters\n",
      "\n",
      "Workflows:\n",
      "  LMM_null\n",
      "  APEX_cis\n",
      "  cis\n",
      "  APEX_trans\n",
      "  trans\n",
      "\n",
      "Global Workflow Options:\n",
      "  --molecular-pheno-list VAL (as path, required)\n",
      "                        Path to the input molecular phenotype file, per chrm, in\n",
      "                        bed.gz format.\n",
      "  --covariate VAL (as path, required)\n",
      "                        Covariate file, in similar format as the molecular_pheno\n",
      "  --genotype-file-list VAL (as path, required)\n",
      "                        Genotype file in vcf format, per chrm\n",
      "  --grm-list VAL (as path, required)\n",
      "                        grm file in pairwise table format , per chrm\n",
      "  --cwd . (as path)\n",
      "                        Path to the work directory of the analysis.\n",
      "  --job-size 2 (as int)\n",
      "                        Specify the number of jobs per run.\n",
      "  --container ''\n",
      "                        Container option for software to run the analysis:\n",
      "                        docker or singularity\n",
      "  --name ROSMAP\n",
      "                        Prefix for the analysis output\n",
      "  --window 1000000 (as list)\n",
      "                        Specify the scanning window for the up and downstream\n",
      "                        radius to analyze around the region of interest, in\n",
      "                        units of bp\n",
      "\n",
      "Sections\n",
      "  APEX_*_1, LMM_null:\n",
      "  APEX_cis_2, cis:\n",
      "  APEX_trans_2, trans:\n",
      "    Workflow Options:\n",
      "      --keep-gene ''\n",
      "  APEX_*_3:\n",
      "  APEX_*_4:\n"
     ]
    }
   ],
   "source": [
    "sos run APEX.ipynb -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Example\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "sos run APEX.ipynb APEX_cis   \\\n",
    "--genotype_file_list GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.vcf_files_list.txt \\\n",
    "--molecular_pheno_list /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.processed_phenotype.per_chrom.recipe   \\\n",
    "--grm_list  data_preprocessing/genotype/grm/plink_files_list.loco_grm_list.txt  \\\n",
    "--covariate data_preprocessing/ALL/covariates/ALL.covariate.pca.BiCV.cov.gz     \\\n",
    "--cwd /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/Association_scan/ALL/APEX    \\\n",
    "--container /mnt/mfs/statgen/containers/apex.sif -J 50 -c csg.yml -q csg -n --no-LMM &"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Global parameter settings\n",
    "The section outlined the parameters that can be set in the command interface."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [],
   "source": [
    "[global]\n",
    "# Path to the input molecular phenotype file, per chrom, in bed.gz format.\n",
    "parameter: phenotype_list = path\n",
    "# Covariate file\n",
    "parameter: covariate_file = path\n",
    "# Genotype file in VCF format, per chrom\n",
    "parameter: genotype_list = path\n",
    "# GRM file in plain text format, per chrom, for leave one chrom out analysis\n",
    "parameter: grm_list = path()\n",
    "# Use LMM or not\n",
    "parameter: LMM = False\n",
    "# Whether or not to apply rank normalization\n",
    "parameter: rankNormal = False\n",
    "# Path to the work directory of the analysis.\n",
    "parameter: cwd = path('.')\n",
    "# Container option for software to run the analysis: docker or singularity\n",
    "parameter: container = ''\n",
    "# Prefix for the analysis output\n",
    "parameter: name = f\"{phenotype_list:bn}_{covariate_file:bn}\"\n",
    "# Specify the scanning window for the up and downstream radius to analyze around the region of interest, in units of bp\n",
    "parameter: window = 1000000\n",
    "# Number of threads\n",
    "parameter: numThreads = 1\n",
    "# For cluster jobs, number commands to run per job\n",
    "parameter: job_size = 1\n",
    "parameter: walltime = '5h'\n",
    "parameter: mem = '80G'\n",
    "\n",
    "import pandas as pd\n",
    "molecular_pheno_chr_inv = pd.read_csv(phenotype_list,sep = \"\\t\")\n",
    "geno_chr_inv = pd.read_csv(genotype_list,sep = \"\\t\")\n",
    "input_inv = molecular_pheno_chr_inv.merge(geno_chr_inv, on = \"#id\")\n",
    "if LMM:\n",
    "    grm_chr_inv =  pd.read_csv(grm_list,sep = \"\\t\")\n",
    "    input_inv = input_inv.merge(grm_chr_inv,on = \"#id\")\n",
    "input_inv = input_inv.values.tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## LMM Regression  \n",
    "This step are done to precompute and store:\n",
    "1. LMM null models and trait residuals\n",
    "2. spline terms for LMM genotypic variances to speed up downstream analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "[LMM_null]\n",
    "input: input_inv, group_by = 4\n",
    "output: f'{cwd:a}/{_input[1]:bnn}.theta.gz'\n",
    "task: trunk_workers = 1, trunk_size=job_size, walltime = walltime, mem = mem, cores = numThreads, tags = f'{step_name}_{_output:bn}'\n",
    "bash: expand= \"$[ ]\", stderr = f'{_output}.stderr', stdout = f'{_output}.stdout', container = container\n",
    "    apex lmm $[\"--rankNormal\" if rankNormal else \"\"] --vcf $[_input[2]] \\\n",
    "    --bed $[_input[1]] \\\n",
    "    --cov $[covariate] \\\n",
    "    --out $[_output[0]:nn] \\\n",
    "    --grm $[_input[3]] \\\n",
    "    --get-theta \\\n",
    "    --threads $[numThreads]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## QTL asscoiation testing \n",
    "This step generate the cis-QTL and trans-QTL summary statistics and for downstream analysis from summary statistics. The analysis is done per chromosome to reduce running time."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "[cis_1]\n",
    "if LMM:\n",
    "    sos_run(\"LMM_null\")\n",
    "input: input_inv, group_by = 4\n",
    "output:f'{cwd:a}/{path(_input[1]):bnn}{\".LMM\" if LMM else \".OLS\"}.cis_long_table.txt.gz',\n",
    "       f'{cwd:a}/{path(_input[1]):bnn}{\".LMM\" if LMM else \".OLS\"}.cis_gene_table.txt.gz',\n",
    "       f'{cwd:a}/{path(_input[1]):bnn}{\".LMM\" if LMM else \".OLS\"}.cis_sumstats.txt.gz'\n",
    "parameter: theta = path(f'{cwd:a}/{_input[1]:bnn}.theta.gz')\n",
    "task: trunk_workers = 1, trunk_size=job_size, walltime = walltime, mem = mem, cores = numThreads, tags = f'{step_name}_{_output:bn}'\n",
    "bash: expand= \"$[ ]\", stderr = f'{_output[0]}.stderr', stdout = f'{_output[0]}.stdout', container = container\n",
    "    apex cis $[\"--rankNormal\" if rankNormal else \"\"] --vcf $[_input[2]] \\\n",
    "    --bed $[_input[1]] \\\n",
    "    --cov $[covariate] \\\n",
    "    --out $[_output[0]:nnn] \\\n",
    "    --long $[f'--theta {theta} --grm {_input[3]}' if LMM else \"\"] \\\n",
    "    --window $[window] \\\n",
    "    --threads $[numThreads]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "[trans_1]\n",
    "if LMM:\n",
    "    sos_run(\"LMM_null\")\n",
    "parameter: keep_gene = \"\"\n",
    "input: input_inv, group_by = 4\n",
    "output: f'{cwd:a}/{_input[1]):bnn}{\".LMM\" if LMM else \".OLS\"}.trans_long_table.txt.gz',\n",
    "        f'{cwd:a}/{_input[1]):bnn}{\".LMM\" if LMM else \".OLS\"}.trans_gene_table.txt.gz'\n",
    "task: trunk_workers = 1, trunk_size=job_size, walltime = walltime, mem = mem, cores = numThreads, tags = f'{step_name}_{_output:bn}'\n",
    "bash: expand= \"$[ ]\", stderr = f'{_output[0]}.stderr', stdout = f'{_output[0]}.stdout', container = container\n",
    "    apex trans $[\"--rankNormal\" if rankNormal else \"\"] --vcf $[_input[2]] \\\n",
    "    --bed $[_input_inv[1]] \\\n",
    "    --cov $[covariate] \\\n",
    "    --out $[_output[0]:nnn] \\\n",
    "    --gene \"$[keep_gene]\" \\\n",
    "    --long $[f'--theta {cwd:a}/{_input[1]:bnn}.theta.gz --grm {_input[3]}' if LMM else \"\"] \\\n",
    "    --threads $[numThreads]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "[*_2]\n",
    "output: f'{_input[0]:nn}.reformated.txt'\n",
    "task: trunk_workers = 1, trunk_size=job_size, walltime = walltime, mem = mem, cores = numThreads, tags = f'{step_name}_{_output:bn}'\n",
    "R: expand = \"$[ ]\", stderr = f'{_output[0]}.stderr', stdout = f'{_output[0]}.stdout',container = container\n",
    "    library(\"dplyr\")\n",
    "    library(\"tibble\")\n",
    "    library(\"readr\")\n",
    "    library(\"purrr\")\n",
    "    data <- read_delim(\"$[_input[0]]\",delim = \"\\t\")\n",
    "    data = data%>%mutate(variant_id = pmap_chr(list(a = `#chrom`,b = pos,c = ref, d = alt),function(a,b,c,d) paste(c(a,\":\",b,\"_\",c,\"_\",d),collapse = \"\")))%>%rename(chrom = `#chrom`)\n",
    "    data %>%write_delim(\"$[_output]\",delim = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "[*_3]\n",
    "input: group_by = \"all\"\n",
    "output: f'{cwd}/APEX_QTL_recipe.tsv',f'{cwd:a}/APEX_column_info.txt'\n",
    "python: expand = \"$[ ]\", stderr = f'{_output[0]}.stderr', stdout = f'{_output[0]}.stdout'\n",
    "    import pandas as pd \n",
    "    data_tempt = pd.DataFrame({\n",
    "    \"#id\" : [int(x.split(\".\")[-6].replace(\"chr\",\"\")) for x in  [$[_input:r,]]],\n",
    "    \"sumstat_dir\" : [$[_input:r,]],\\\n",
    "    \"column_info\" : $[_output[1]:r]\n",
    "    })\n",
    "    column_info_df = pd.DataFrame( pd.Series( {\"ID\": \"GENE,CHR,POS,A0,A1\",\n",
    "          \"CHR\": \"chrom\",\n",
    "          \"POS\": \"pos\",\n",
    "          \"A0\": \"ref\",\n",
    "          \"A1\": \"alt\",\n",
    "          \"SNP\": \"variant_id\",\n",
    "          \"STAT\": \"beta\",\n",
    "          \"SE\": \"se\",\n",
    "          \"P\": \"pval\",\n",
    "          \"GENE\": \"gene\"}), columns = [\"APEX\"] )\n",
    "    data_tempt.to_csv(\"$[_output[0]]\",index = False,sep = \"\\t\" )\n",
    "    column_info_df.to_csv(\"$[_output[1]]\",index = True,sep = \"\\t\" )"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "Bash",
     "#E6EEFF",
     ""
    ],
    [
     "SoS",
     "sos",
     "",
     "",
     "sos"
    ]
   ],
   "version": "0.22.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
