
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17: http://docutils.sourceforge.net/" />

    <title>Bulk RNA-seq eQTL analysis &#8212; ADSP-FG Project</title>
    
  <link href="../../_static/css/theme.css" rel="stylesheet">
  <link href="../../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/togglebutton.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script>
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Reference data standardization" href="../data_preprocessing/reference_data.html" />
    <link rel="prev" title="RNA-seq calling and QC" href="bulk_expression_commands.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../../_static/xqtl_wf.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">ADSP-FG Project</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../README.html">
   xQTL analysis pipeline
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Command Generator
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="bulk_expression_commands.html">
   RNA-seq calling and QC
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Bulk RNA-seq eQTL analysis
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Reference data
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../data_preprocessing/reference_data.html">
   Reference data standardization
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Molecular Phenotypes
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../molecular_phenotypes/bulk_expression.html">
   RNA-seq calling and QC
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../molecular_phenotypes/calling/RNA_calling.html">
     Quantifying expression from RNA-seq data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../molecular_phenotypes/QC/bulk_expression_QC.html">
     Sample level RNA-seq quality control
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../molecular_phenotypes/QC/bulk_expression_normalization.html">
     Bulk RNA-seq counts normalization
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../molecular_phenotypes/scnuc_expression.html">
   scRNA-seq expression calling
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../molecular_phenotypes/apa.html">
   APA calling
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../molecular_phenotypes/calling/apa_calling.html">
     APA Calling
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../molecular_phenotypes/splicing.html">
   Splicing events calling
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../molecular_phenotypes/calling/splicing_calling.html">
     Quantifying alternative splicing from RNA-seq data
    </a>
   </li>
  </ul>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Data Pre-processing
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../data_preprocessing/genotype_preprocessing.html">
   Genotype data preprocessing
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/genotype/VCF_QC.html">
     Genotype VCF file quality control
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/genotype/GWAS_QC.html">
     Genotype PLINK file quality control
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/genotype/PCA.html">
     Principal component analysis
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/genotype/GRM.html">
     Genomic relationship matrices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/genotype/genotype_formatting.html">
     Genotype data formatting
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../data_preprocessing/phenotype_preprocessing.html">
   Phenotype data preprocessing
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/phenotype/gene_annotation.html">
     Gene coordinate annotation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/phenotype/phenotype_formatting.html">
     Phenotype data formatting
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../data_preprocessing/covariate_preprocessing.html">
   Covariate data preprocessing
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/covariate/PEER_factor.html">
     Factor analysis using PEER
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/covariate/BiCV_factor.html">
     Factor analysis using Bi-Cross validation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../data_preprocessing/covariate/covariate_formatting.html">
     Merge Covariate
    </a>
   </li>
  </ul>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  QTL Association Testing
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../association_scan/cisQTL_scan.html">
   cisQTL analysis workflows
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../association_scan/TensorQTL/TensorQTL.html">
     TensorQTL QTL association testing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../association_scan/APEX/APEX.html">
     APEX QTL association testing
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../association_scan/transQTL_scan.html">
   transQTL analysis workflows
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Multivariate Analysis
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../multivariate/METAL_pipeline.html">
   Meta-analysis with METAL
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-8" name="toctree-checkbox-8" type="checkbox"/>
  <label for="toctree-checkbox-8">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../multivariate/METAL/METAL.html">
     Meta-analysis with METAL
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../multivariate/MASH_pipeline.html">
   Multivariate association with MASH
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-9" name="toctree-checkbox-9" type="checkbox"/>
  <label for="toctree-checkbox-9">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../multivariate/MASH/fastqtl_to_mash.html">
     eQTL summary statistics formatting
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../multivariate/MASH/mixture_prior.html">
     A multivariate EBNM approach for mixture multivariate distribution estimate
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../multivariate/MASH/mashr.html">
     MASH analysis pipeline with data-driven prior matrices
    </a>
   </li>
  </ul>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Statistical Fine-mapping
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../fine_mapping/univariate_fine_mapping.html">
   Univariate fine-mapping workflows
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-10" name="toctree-checkbox-10" type="checkbox"/>
  <label for="toctree-checkbox-10">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../fine_mapping/SuSiE/SuSiE_RSS.html">
     Fine-mapping with SuSiE RSS model
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../fine_mapping/SuSiE/SuSiE.html">
     Fine-mapping with SuSiE model
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../fine_mapping/multivariate_fine_mapping.html">
   Multivariate fine-mapping workflows
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Multi-omics integration
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../integrative_analysis/colocalization.html">
   QTL and GWAS Colocalization
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-11" name="toctree-checkbox-11" type="checkbox"/>
  <label for="toctree-checkbox-11">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../integrative_analysis/COLOC/fastenloc.html">
     Colocalization with FastENLOC
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../integrative_analysis/COLOC/mvenloc.html">
     Multivariate SuSiE and ENLOC model
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../integrative_analysis/functional_annotation.html">
   Functional annotation integration
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-12" name="toctree-checkbox-12" type="checkbox"/>
  <label for="toctree-checkbox-12">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../integrative_analysis/enrichment/gwas_enrichment.html">
     Enrichment analysis: TORUS and GREGOR
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../integrative_analysis/enrichment/ldsc.html">
     Stratified LD Score Regression
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../integrative_analysis/SuSiE_Ann/polyfun.html">
     Fine-mapping with PolyFun
    </a>
   </li>
  </ul>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Utilities
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../misc/summary_stats_merger.html">
   Summary statistics merger
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../misc/rds_to_vcf.html">
   Summary statistics in VCF format
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Supported by <a href="https://www.nia.nih.gov/research/ad-genetics">NIH/NIA</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../_sources/code/commands_generator/eQTL_analysis_commands.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        
        <a class="issues-button"
            href="https://github.com/cumc/xqtl-pipeline/issues/new?title=Issue%20on%20page%20%2Fcode/commands_generator/eQTL_analysis_commands.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/cumc/xqtl-pipeline/edit/gh-pages/code/commands_generator/eQTL_analysis_commands.ipynb"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#example-for-running-the-workflow">
   Example for running the workflow
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#data-preprocessing">
   Data Preprocessing
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genotype-preprocessing-once-for-all-tissues">
     Genotype Preprocessing (Once for all tissues)
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#molecular-phenotype-processing">
     Molecular Phenotype Processing
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genotype-processing">
     Genotype Processing
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#factor-analysis">
   Factor analysis
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#association-scan">
   Association scan
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#susie">
   SuSiE
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sumstat-merger">
   Sumstat Merger
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="bulk-rna-seq-eqtl-analysis">
<h1>Bulk RNA-seq eQTL analysis<a class="headerlink" href="#bulk-rna-seq-eqtl-analysis" title="Permalink to this headline">¶</a></h1>
<p>This notebook provide a master control on the XQTL workflow so it can automate the work before data preprocessing on multiple data collection as proposed.</p>
<p>This master control notebook is mainly to serve the 8 tissues snuc_bulk_expression analysis, but should be functional on all analysis where expression data an</p>
<p>Input:
A recipe file,each row is a data collection and with the following column:</p>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>Theme
    name of dataset, must be different, each uni_study analysis will be performed in a folder named after each, meta analysis will be performed in a folder named as {study1}_{study2}
        
    The column name must contain the # and be the first column

genotype_file
    {Path to a whole genome genotype file}

molecular_pheno
    {Path to file}
    
covariate_file
    {Path to file}

### note: Only data collection from the same Populations and conditions will me merged to perform Fix effect meta analysis
</pre></div>
</div>
<p>Output:
…</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>nohup sos run ~/GIT/xqtl-pipeline/code/complete_analysis/eQTL_analysis_externalBed.ipynb factor --recipe /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt -s build <span class="p">&amp;</span>

nohup sos dryrun ~/GIT/xqtl-pipeline/code/complete_analysis/eQTL_analysis_externalBed.ipynb TensorQTL <span class="se">\</span>
    --recipe /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new     <span class="se">\</span>
    --genotype_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt      <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt -s build <span class="p">&amp;</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>/mnt/mfs/ctcn/team/masashi/snuc-eqtl/v20211109.celltypes/*/covariates-20211118.tsv
</pre></div>
</div>
</div>
</div>
<section id="example-for-running-the-workflow">
<h2>Example for running the workflow<a class="headerlink" href="#example-for-running-the-workflow" title="Permalink to this headline">¶</a></h2>
<p>This will run the workflow from via several submission and save the output to nohup.out</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sos run ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb TensorQTL <span class="se">\</span>
    --recipe  &lt;<span class="o">(</span>cat /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new  <span class="p">|</span> head -2<span class="o">)</span> <span class="se">\</span>
    --genotype_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt      <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt -n <span class="p">&amp;</span>

sos run ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb APEX <span class="se">\</span>
    --recipe  &lt;<span class="o">(</span>cat /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new  <span class="p">|</span> head -2<span class="o">)</span> <span class="se">\</span>
    --genotype_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt      <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt -n <span class="p">&amp;</span>


sos run ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb sumstat_merge  <span class="se">\</span>
    --recipe  &lt;<span class="o">(</span>cat /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new  <span class="p">|</span> head -2<span class="o">)</span>    <span class="se">\</span>
    --genotype_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt      <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt --Association_option <span class="s2">&quot;TensorQTL&quot;</span>   -n


sos run ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb sumstat_merge <span class="se">\</span>
    --recipe  &lt;<span class="o">(</span>cat /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new  <span class="p">|</span> head -2<span class="o">)</span> <span class="se">\</span>
    --genotype_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt      <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt -n



sos run ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb APEX <span class="se">\</span>
    --recipe /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/recipe_ALL_celltype     <span class="se">\</span>
    --genotype_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt      <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt   <span class="p">&amp;</span>


nohup sos run /home/hs3163/GIT/xqtl-pipeline/pipeline//genotype_formatting.ipynb plink_by_gene <span class="se">\</span>
    --genoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.bed <span class="se">\</span>
    --cwd /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis <span class="se">\</span>
    --region_list  /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.region_list <span class="se">\</span>
    --container /mnt/mfs/statgen/containers/bioinfo.sif  -J <span class="m">500</span> -c csg.yml -q csg

sos dryrun ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb plink_by_gene <span class="se">\</span>
    --recipe /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new     <span class="se">\</span>
    --ref_fasta /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/GRCh38_full_analysis_set_plus_decoy_hla.noALT_noHLA_noDecoy_ERCC.fasta    <span class="se">\</span>
    --genoFile  /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_data/GRCh38_liftedover_sorted_all.vcf.gz <span class="se">\</span>
    --dbSNP_vcf  /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/00-All.vcf.gz <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt  &gt; genotype_script <span class="p">&amp;</span>

sos dryrun ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb UniSuSiE <span class="se">\</span>
    --recipe /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new     <span class="se">\</span>
    --genoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.bed <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt 


sos run ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb UniSuSiE <span class="se">\</span>
    --recipe &lt;<span class="o">(</span>cat /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new  <span class="p">|</span> head -2<span class="o">)</span>      <span class="se">\</span>
    --genoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.bed <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt -n 



 Eight_tissue_scripts <span class="p">&amp;</span>



sos dryrun ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb plink_per_chrom <span class="se">\</span>
    --recipe /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new     <span class="se">\</span>
    --ref_fasta /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/GRCh38_full_analysis_set_plus_decoy_hla.noALT_noHLA_noDecoy_ERCC.fasta    <span class="se">\</span>
    --genoFile  /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_data/GRCh38_liftedover_sorted_all.vcf.gz <span class="se">\</span>
    --dbSNP_vcf  /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/00-All.vcf.gz <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt  &gt; genotype_script <span class="p">&amp;</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sos dryrun /home/hs3163/GIT/xqtl-pipeline/pipeline/genotype_formatting.ipynb plink_by_gene <span class="se">\</span>
    --genoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.bed <span class="se">\</span>
    --cwd /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis <span class="se">\</span>
    --region_list  /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.region_list <span class="se">\</span>
    --container /mnt/mfs/statgen/containers/bioinfo.sif  -J <span class="m">500</span> -c csg.yml -q csg


sos dryrun /home/hs3163/GIT/xqtl-pipeline/pipeline/APEX.ipynb APEX_cis <span class="se">\</span>
    --genotype_file_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt   <span class="se">\</span>
    --molecular_pheno_list /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.processed_phenotype.per_chrom.recipe <span class="se">\</span>
    --grm_list  /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/genotype/grm/plink_files_list.loco_grm_list.txt  <span class="se">\</span>
    --covariate data_preprocessing/ALL/covariates/ALL.covariate.pca.BiCV.cov.gz    <span class="se">\</span>
    --cwd /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/Association_scan/ALL/APEX  <span class="se">\</span>
    --container /mnt/mfs/statgen/containers//APEX.sif

 nohup sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/APEX.ipynb APEX_cis --genotype_file_list /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.vcf_files_list.txt --molecular_pheno_list /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.processed_phenotype.per_chrom.recipe --grm_list data_preprocessing/genotype/grm/plink_files_list.loco_grm_list.txt --covariate data_preprocessing/ALL/covariates/ALL.covariate.pca.BiCV.cov.gz --cwd /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/Association_scan/ALL/APEX --container /mnt/mfs/statgen/containers/apex.sif -J <span class="m">50</span> -c csg.yml -q csg
    
    
    
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/SuSiE.ipynb uni_susie    <span class="se">\</span>
    --genoFile &lt;<span class="o">(</span>head -2 /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/ALL.log2cpm_plink_files/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.plink_files_list.txt<span class="o">)</span>   <span class="se">\</span>
    --phenoFile /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.gz  <span class="se">\</span>
    --covFile data_preprocessing/ALL/covariates/ALL.log2cpm.resid.PEER.cov.gz    <span class="se">\</span>
    --cwd /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/Fine_mapping/UniSuSiE   <span class="se">\</span>
    --tissues ALL   <span class="se">\</span>
    --region-list &lt;<span class="o">(</span>head /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.region_list<span class="o">)</span> <span class="se">\</span>
    --container /mnt/mfs/statgen/containers/stephenslab.sif -J <span class="m">50</span> -c csg.yml -q csg -n <span class="p">&amp;</span> 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>apex cis --vcf /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.filtered.chr21.vcf.gz <span class="se">\</span>
--bed /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.chr21.mol_phe.bed.gz <span class="se">\</span>
--cov data_preprocessing/ALL/covariates/ALL.covariate.pca.BiCV.cov.gz <span class="se">\</span>
--out /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/Association_scan/test_apex/ALL.log2cpm.bed.chr21.mol_phe.OLS_winSpecified <span class="se">\</span>
--long  <span class="se">\</span>
--window <span class="s1">&#39;1000000&#39;</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sos run ~/GIT/xqtl-pipeline/pipeline/eQTL_analysis_commands.ipynb sumstat_merge  <span class="se">\</span>
    --recipe  &lt;<span class="o">(</span>cat /mnt/mfs/statgen/snuc_pseudo_bulk//data/recipe_8tissue_new  <span class="p">|</span> head -2<span class="o">)</span>    <span class="se">\</span>
    --genotype_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt      <span class="se">\</span>
    --annotation_gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf     <span class="se">\</span>
    --sample_participant_lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt <span class="se">\</span>
    --factor_option <span class="s2">&quot;PEER&quot;</span> --Association_option <span class="s2">&quot;TensorQTL&quot;</span>   -n
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>INFO: Checking <span class=" -Color -Color-Green">sample_match</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/sample_matcher.ipynb filtered_sample_list \
    --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL  \
    --phenoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/phenotype_data_all/ALL.log2cpm.tsv  \
    --genoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.fam  \
    --sample-participant-lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt \
    --container /mnt/mfs/statgen/containers/rna_quantification.sif \
    --translated_phenoFile 


INFO: <span class=" -Color -Color-Green">sample_match</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">sample_match</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/sampleSheetAfterQC.filtered.txt /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/sampleSheetAfterQC.filtered_geno.txt</span>
INFO: Checking <span class=" -Color -Color-Green">annotation</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/gene_annotation.ipynb annotate_coord \
  --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data \
  --phenoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/phenotype_data_all/ALL.log2cpm.tsv \
  --annotation-gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf \
  --sample-participant-lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt \
  --container /mnt/mfs/statgen/containers/rna_quantification.sif \
  --phenotype-id-type gene_name 


INFO: <span class=" -Color -Color-Green">annotation</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">annotation</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.gz</span>
INFO: Checking <span class=" -Color -Color-Green">king</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/GWAS_QC.ipynb king \
   --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data \
   --genoFile /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.bed \
   --name ALL \
   --keep-samples /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/sampleSheetAfterQC.filtered_geno.txt \
   --container /mnt/mfs/statgen/containers/bioinfo.sif \
   --walltime 48h  


INFO: <span class=" -Color -Color-Green">king</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">king</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.bed /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.bed</span>
INFO: Checking <span class=" -Color -Color-Green">region_list_generation</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/gene_annotation.ipynb region_list_generation \
    --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data  \
    --phenoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.gz\
    --annotation-gtf /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/genes.reformatted.gene.gtf \
    --sample-participant-lookup /mnt/mfs/statgen/snuc_pseudo_bulk/data/reference_data/sampleSheetAfterQC.txt \
    --container /mnt/mfs/statgen/containers/rna_quantification.sif \
    --phenotype-id-type gene_name 


INFO: <span class=" -Color -Color-Green">region_list_generation</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">region_list_generation</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.region_list</span>
INFO: Checking <span class=" -Color -Color-Green">unrelated_QC</span>: 
INFO: Checking <span class=" -Color -Color-Green">phenotype_partition_by_chrom</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/GWAS_QC.ipynb qc \
   --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data \
   --genoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.bed \
   --exclude-variants /mnt/mfs/statgen/snuc_pseudo_bulk/Ast/genotype/dupe_snp_to_exclude \
   --maf-filter 0.05 \
   --container /mnt/mfs/statgen/containers/bioinfo.sif \
   --mem 40G   


INFO: <span class=" -Color -Color-Green">unrelated_QC</span> is <span class=" -Color -Color-Green">completed</span>.
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/phenotype_formatting.ipynb partition_by_chrom \
    --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data  \
    --phenoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.gz \
    --region-list /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.region_list \
    --container /mnt/mfs/statgen/containers/rna_quantification.sif \
    --mem 4G      


INFO: <span class=" -Color -Color-Green">phenotype_partition_by_chrom</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">unrelated_QC</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.bed /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.in</span>
INFO: <span class=" -Color -Color-Green">phenotype_partition_by_chrom</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.processed_phenotype.per_chrom.recipe</span>
INFO: Checking <span class=" -Color -Color-Green">pca</span>: 
INFO: Checking <span class=" -Color -Color-Green">related_QC</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/PCA.ipynb flashpca \
   --cwd data_preprocessing/ALL/pca data_preprocessing/ALL/pca \
   --genoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.bed \
   --container /mnt/mfs/statgen/containers/flashpcaR.sif  


INFO: <span class=" -Color -Color-Green">pca</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">pca</span> output:   <span class=" -Color -Color-Green">data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.pca.rds data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.pca.scree.txt</span>
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/GWAS_QC.ipynb qc_no_prune \
   --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data \
   --genoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.bed \
   --exclude-variants /mnt/mfs/statgen/snuc_pseudo_bulk/Ast/genotype/dupe_snp_to_exclude \
   --maf-filter 0 \
   --geno-filter 0 \
   --mind-filter 0.1 \
   --hwe-filter 0 \
   --keep-variants /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.in  \
   --container /mnt/mfs/statgen/containers/bioinfo.sif  \
   --mem 40G   


INFO: <span class=" -Color -Color-Green">related_QC</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">related_QC</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.filtered.extracted.bed</span>
INFO: Checking <span class=" -Color -Color-Green">projected_sample</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/PCA.ipynb project_samples \
        --cwd data_preprocessing/ALL/pca data_preprocessing/ALL/pca \
        --genoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.filtered.extracted.bed \
        --pca-model  data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.pca.rds \
        --maha-k `awk &#39;$3 &lt; 0.7&#39; data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.unrelated.filtered.prune.pca.scree.txt | tail -1 | cut -f 1  ` \
        --container /mnt/mfs/statgen/containers/flashpcaR.sif  


INFO: <span class=" -Color -Color-Green">projected_sample</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">projected_sample</span> output:   <span class=" -Color -Color-Green">data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.filtered.extracted.pca.projected.rds data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.filtered.extracted.pca.projected.scree.txt</span>
INFO: Checking <span class=" -Color -Color-Green">merge_pca_covariate</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/covariate_formatting.ipynb merge_pca_covariate \
        --cwd data_preprocessing/ALL/covariates \
        --pcaFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.filtered.extracted.pca.projected.rds /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.filtered.extracted.pca.projected.scree.txt \
        --covFile  /mnt/mfs/statgen/snuc_pseudo_bulk/data/covariate_data/ALL.covariate.tsv \
        --tol_cov 0.3  \
        --k `awk &#39;$3 &lt; 0.7&#39; data_preprocessing/ALL/pca/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related.filtered.extracted.pca.projected.scree.txt | tail -1 | cut -f 1 `
        --container /mnt/mfs/statgen/containers/bioinfo.sif 


INFO: <span class=" -Color -Color-Green">merge_pca_covariate</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">merge_pca_covariate</span> output:   <span class=" -Color -Color-Green">data_preprocessing/ALL/covariates/ALL.covariate.pca.cov.gz</span>
INFO: Checking <span class=" -Color -Color-Green">resid_exp</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/covariate_formatting.ipynb Residual_Y \
        --cwd data_preprocessing/ALL/phenotype \
        --phenoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.gz \
        --covFile  /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/covariates/ALL.covariate.pca.cov.gz \
        --container /mnt/mfs/statgen/containers/bioinfo.sif 


INFO: <span class=" -Color -Color-Green">resid_exp</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">resid_exp</span> output:   <span class=" -Color -Color-Green">data_preprocessing/ALL/phenotype/ALL.log2cpm.resid.bed.gz</span>
INFO: Checking <span class=" -Color -Color-Green">factor</span>: 
script_6697980166679677035:4: SyntaxWarning: &quot;is&quot; with a literal. Did you mean &quot;==&quot;?
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/PEER_factor.ipynb PEER \
   --cwd data_preprocessing/ALL/covariates \
   --phenoFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype/ALL.log2cpm.resid.bed.gz  \
   --container /mnt/mfs/statgen/containers/PEER.sif  \
   --walltime 24h \
   --numThreads 8 \
   --iteration 1000 \
   --N 60  


INFO: <span class=" -Color -Color-Green">factor</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">factor</span> output:   <span class=" -Color -Color-Green">data_preprocessing/ALL/covariates/ALL.log2cpm.resid.PEER.cov.gz</span>
INFO: Checking <span class=" -Color -Color-Green">merge_factor_covariate</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/covariate_formatting.ipynb merge_factor_covariate \
        --cwd data_preprocessing/ALL/covariates \
        --factorFile /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/covariates/ALL.log2cpm.resid.PEER.cov.gz \
        --covFile  /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/covariates/ALL.covariate.pca.cov.gz \
        --container /mnt/mfs/statgen/containers/bioinfo.sif 


INFO: <span class=" -Color -Color-Green">merge_factor_covariate</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">merge_factor_covariate</span> output:   <span class=" -Color -Color-Green">data_preprocessing/ALL/covariates/ALL.covariate.pca.PEER.cov.gz</span>
INFO: Checking <span class=" -Color -Color-Green">TensorQTL</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/TensorQTL.ipynb TensorQTL_cis \
    --genotype_file_list /mnt/mfs/statgen/snuc_pseudo_bulk/data/genotype_qced/plink_files_list.txt   \
    --molecular_pheno_list /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/data_preprocessing/ALL/phenotype_data/ALL.log2cpm.bed.processed_phenotype.per_chrom.recipe \
    --covariate data_preprocessing/ALL/covariates/ALL.covariate.pca.PEER.cov.gz    \
    --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Association_scan/ALL/TensorQTL  \
    --container /mnt/mfs/statgen/containers//TensorQTL.sif 


INFO: <span class=" -Color -Color-Green">TensorQTL</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">TensorQTL</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Association_scan/ALL/TensorQTL/TensorQTL_recipe.tsv</span>
INFO: Checking <span class=" -Color -Color-Green">yml_generation</span>: 
HINT: report:
sos run /home/hs3163/GIT/xqtl-pipeline/pipeline/yml_generator.ipynb yml_list \
    --sumstat-list /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Association_scan/ALL/TensorQTL/TensorQTL_recipe.tsv \
    --cwd  /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration --name ALL 




INFO: <span class=" -Color -Color-Green">yml_generation</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: <span class=" -Color -Color-Green">yml_generation</span> output:   <span class=" -Color -Color-Green">/mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration/qced_sumstat_list.txt /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration/TensorQTL/yml_list.txt</span>
INFO: Checking <span class=" -Color -Color-Green">sumstat_merge</span>: 
HINT: report:
sos run  /home/hs3163/GIT/xqtl-pipeline/pipeline/summary_stats_merger.ipynb     \
      --sumstat-list /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration/qced_sumstat_list.txt    \
      --yml-list /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration/TensorQTL/yml_list.txt    \
      --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration   --mem 50G --walltime 48h 

sos run  /home/hs3163/GIT/xqtl-pipeline/pipeline/summary_stats_merger.ipynb    sumstat_to_vcf  \
   --sumstat-list /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration/qced_sumstat_list.txt    \
   --yml-list /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration/TensorQTL/yml_list.txt    \
   --cwd /mnt/mfs/hgrcgrid/homes/hs3163/GIT/xqtl-pipeline/code/commands_generator/Data_intergration   --mem 50G --walltime 48h 



INFO: <span class=" -Color -Color-Green">sumstat_merge</span> is <span class=" -Color -Color-Green">completed</span>.
INFO: Workflow sumstat_merge (ID=wb37e2362b1536bb9) is tested successfully with 16 completed steps.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>global<span class="o">]</span>
<span class="c1">## The aforementioned input recipe</span>
parameter: <span class="nv">recipe</span> <span class="o">=</span> path
<span class="c1">## Overall wd, the file structure of analysis is wd/[steps]/[sub_dir for each steps]</span>
parameter:  <span class="nv">cwd</span> <span class="o">=</span> path<span class="o">(</span><span class="s2">&quot;output&quot;</span><span class="o">)</span>
<span class="c1">## Diretory to the excutable</span>
parameter: <span class="nv">exe_dir</span> <span class="o">=</span> path<span class="o">(</span><span class="s2">&quot;~/GIT/xqtl-pipeline/&quot;</span><span class="o">)</span>
parameter: <span class="nv">container</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers/twas_latest.sif&#39;</span>
parameter: <span class="nv">container_base_bioinfo</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers/bioinfo.sif&#39;</span>
parameter: <span class="nv">container_apex</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers/apex.sif&#39;</span>
parameter: <span class="nv">container_PEER</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers/PEER.sif&#39;</span>
parameter: <span class="nv">container_TensorQTL</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers//TensorQTL.sif&#39;</span>
parameter: <span class="nv">container_rnaquant</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers/rna_quantification.sif&#39;</span>
parameter: <span class="nv">container_flashpca</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers/flashpcaR.sif&#39;</span>
parameter: <span class="nv">container_susie</span> <span class="o">=</span> <span class="s1">&#39;/mnt/mfs/statgen/containers/stephenslab.sif&#39;</span>
parameter: <span class="nv">sample_participant_lookup</span> <span class="o">=</span> path
parameter: <span class="nv">phenotype_id_type</span> <span class="o">=</span> <span class="s2">&quot;gene_name&quot;</span> 
parameter: <span class="nv">yml</span> <span class="o">=</span> path<span class="o">(</span><span class="s2">&quot;csg.yml&quot;</span><span class="o">)</span>
import pandas as pd
<span class="nv">input_inv</span> <span class="o">=</span> pd.read_csv<span class="o">(</span>recipe, <span class="nv">sep</span> <span class="o">=</span> <span class="s2">&quot;\t&quot;</span><span class="o">)</span>.to_dict<span class="o">(</span><span class="s2">&quot;records&quot;</span><span class="o">)</span>
import os
parameter: <span class="nb">jobs</span> <span class="o">=</span> <span class="m">50</span> <span class="c1"># Number of jobs that are submitted to the cluster</span>
parameter: <span class="nv">queue</span> <span class="o">=</span> <span class="s2">&quot;csg&quot;</span> <span class="c1"># The queue that jobs are submitted to</span>
<span class="nv">submission</span> <span class="o">=</span> f<span class="s1">&#39;-J {jobs} -c {yml} -q {queue}&#39;</span>


<span class="c1">## Control of the workflow</span>
<span class="c1">### Factor option (PEER vs BiCV)</span>
parameter: <span class="nv">factor_option</span> <span class="o">=</span> <span class="s2">&quot;PEER&quot;</span>
<span class="c1">### Association scan option (APEX vs TensorQTL)</span>
parameter: <span class="nv">Association_option</span> <span class="o">=</span> <span class="s2">&quot;TensorQTL&quot;</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="data-preprocessing">
<h2>Data Preprocessing<a class="headerlink" href="#data-preprocessing" title="Permalink to this headline">¶</a></h2>
<section id="genotype-preprocessing-once-for-all-tissues">
<h3>Genotype Preprocessing (Once for all tissues)<a class="headerlink" href="#genotype-preprocessing-once-for-all-tissues" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>dbSNP<span class="o">]</span>
parameter: <span class="nv">dbSNP_vcf</span> <span class="o">=</span> path
input: dbSNP_vcf
output: f<span class="s1">&#39;{cwd:a}/{_input:bnn}.add_chr.variants.gz&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline//VCF_QC.ipynb dbsnp_annotate <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>VCF_add_chr<span class="o">]</span>
parameter: <span class="nv">genoFile</span> <span class="o">=</span> path
input: genoFile
output: f<span class="s1">&#39;{cwd:a}/{_input:bnn}.add_chr.vcf.gz&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline//VCF_QC.ipynb rename_chrs <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span> <span class="se">\</span>
            --walltime <span class="s2">&quot;24h&quot;</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>VCF_QC<span class="o">]</span>
parameter: <span class="nv">ref_fasta</span> <span class="o">=</span> path
input: output_from<span class="o">(</span><span class="s2">&quot;VCF_add_chr&quot;</span><span class="o">)</span>, output_from<span class="o">(</span><span class="s2">&quot;dbSNP&quot;</span><span class="o">)</span>
output: f<span class="s1">&#39;{_input:nn}.leftnorm.filtered.bed&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>//pipeline/VCF_QC.ipynb rename_chrs <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span> <span class="se">\</span>
            --dbsnp-variants $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span>
            --reference-genome $<span class="o">[</span>ref_fasta<span class="o">]</span> <span class="se">\ </span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span> <span class="se">\</span>
            --walltime <span class="s2">&quot;24h&quot;</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>plink_QC<span class="o">]</span>
<span class="c1"># minimum MAF filter to use. 0 means do not apply this filter.</span>
parameter: <span class="nv">maf_filter</span> <span class="o">=</span> <span class="m">0</span>.0
<span class="c1"># maximum MAF filter to use. 0 means do not apply this filter.</span>
parameter: <span class="nv">maf_max_filter</span> <span class="o">=</span> <span class="m">0</span>.0
<span class="c1"># Maximum missingess per-variant</span>
parameter: <span class="nv">geno_filter</span> <span class="o">=</span> <span class="m">0</span>.1
<span class="c1"># Maximum missingness per-sample</span>
parameter: <span class="nv">mind_filter</span> <span class="o">=</span> <span class="m">0</span>.1
<span class="c1"># HWE filter </span>
parameter: <span class="nv">hwe_filter</span> <span class="o">=</span> 1e-06
input: output_from<span class="o">(</span><span class="s2">&quot;VCF_QC&quot;</span><span class="o">)</span>
output: f<span class="s1">&#39;{_input:n}.filtered.bed&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>//pipeline/GWAS_QC.ipynb qc_no_prune <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
            --maf-filter $<span class="o">[</span>maf_filter<span class="o">]</span> <span class="se">\</span>
            --geno-filter $<span class="o">[</span>geno_filter<span class="o">]</span> <span class="se">\</span>
            --mind-filter $<span class="o">[</span>mind_filter<span class="o">]</span> <span class="se">\</span>
            --hwe-filter $<span class="o">[</span>hwe_filter<span class="o">]</span>   <span class="se">\</span>
            --mem 40G  <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>plink_per_chrom<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;plink_QC&quot;</span><span class="o">)</span>
output: f<span class="s1">&#39;{cwd:a}/{_input:bn}.plink_files_list.txt&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
        <span class="k">for</span> i <span class="k">in</span> <span class="o">{</span><span class="m">1</span>..22<span class="o">}</span> <span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> chr<span class="nv">$i</span><span class="p">;</span> <span class="k">done</span> &gt; vcf_chr_list
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>//pipeline/genotype_formatting.ipynb plink_by_chrom <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --chrom  <span class="sb">`</span>cat vcf_chr_list<span class="sb">`</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
        rm vcf_chr_list
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>plink_to_vcf<span class="o">]</span>
parameter: <span class="nv">genotype_list</span> <span class="o">=</span> path
input: genotype_list
output: f<span class="s1">&#39;{cwd:a}/{_input:bnn}.vcf_files_list.txt&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>//pipeline/genotype_formatting.ipynb plink_to_vcf <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>plink_per_gene<span class="o">]</span>
<span class="c1"># The plink genotype file</span>
parameter: <span class="nv">genoFile</span> <span class="o">=</span> path
input: output_from<span class="o">(</span><span class="s2">&quot;region_list_concat&quot;</span><span class="o">)</span>,genoFile
output: f<span class="s1">&#39;{cwd:a}/{_input[1]:bn}.plink_files_list.txt&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline//genotype_formatting.ipynb plink_by_gene <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span> <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --region_list  $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="molecular-phenotype-processing">
<h3>Molecular Phenotype Processing<a class="headerlink" href="#molecular-phenotype-processing" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>annotation<span class="o">]</span>
<span class="c1">## Must be ran with internet connection</span>
import os
parameter: <span class="nv">annotation_gtf</span> <span class="o">=</span> path
input: <span class="nv">for_each</span> <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd:a}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/phenotype_data/{path(_input_inv[&quot;molecular_pheno&quot;]):bn}.bed.gz&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
  sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/gene_annotation.ipynb annotate_coord <span class="se">\</span>
    --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
    --phenoFile $<span class="o">[</span>_input_inv<span class="o">[</span><span class="s2">&quot;molecular_pheno&quot;</span><span class="o">]]</span> <span class="se">\</span>
    --annotation-gtf $<span class="o">[</span>annotation_gtf<span class="o">]</span> <span class="se">\</span>
    --sample-participant-lookup $<span class="o">[</span>sample_participant_lookup<span class="o">]</span> <span class="se">\</span>
    --container $<span class="o">[</span>container_rnaquant<span class="o">]</span> <span class="se">\</span>
    --phenotype-id-type $<span class="o">[</span>phenotype_id_type<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>region_list_generation<span class="o">]</span>
parameter: <span class="nv">annotation_gtf</span> <span class="o">=</span> path
input: output_from<span class="o">(</span><span class="s2">&quot;annotation&quot;</span><span class="o">)</span>, <span class="nv">group_with</span> <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: <span class="nv">pheno_mod</span> <span class="o">=</span> f<span class="s1">&#39;{cwd:a}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/phenotype_data/{_input:bnn}.region_list&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/gene_annotation.ipynb region_list_generation <span class="se">\</span>
        --cwd $<span class="o">[</span>_output:d<span class="o">]</span>  <span class="se">\</span>
        --phenoFile $<span class="o">[</span>_input<span class="o">]</span><span class="se">\</span>
        --annotation-gtf $<span class="o">[</span>annotation_gtf<span class="o">]</span> <span class="se">\</span>
        --sample-participant-lookup $<span class="o">[</span>sample_participant_lookup<span class="o">]</span> <span class="se">\</span>
        --container $<span class="o">[</span>container_rnaquant<span class="o">]</span> <span class="se">\</span>
        --phenotype-id-type $<span class="o">[</span>phenotype_id_type<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>region_list_concat<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;region_list_generation&quot;</span><span class="o">)</span>, <span class="nv">group_by</span> <span class="o">=</span> <span class="s2">&quot;all&quot;</span>
output: f<span class="s1">&#39;{cwd:a}/data_preprocessing/phenotype_data/concat.region_list&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
         cat $<span class="o">[</span>_input:a<span class="o">]</span> <span class="p">|</span> sort <span class="p">|</span> uniq &gt; $<span class="o">[</span>_output:a<span class="o">]</span> 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>phenotype_partition_by_chrom<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;annotation&quot;</span><span class="o">)</span>,output_from<span class="o">(</span><span class="s2">&quot;region_list_generation&quot;</span><span class="o">)</span>, <span class="nv">group_with</span> <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: <span class="nv">per_chrom_pheno_list</span> <span class="o">=</span> f<span class="s1">&#39;{cwd:a}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/phenotype_data/{_input[0]:bn}.processed_phenotype.per_chrom.recipe&#39;</span>        
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/phenotype_formatting.ipynb partition_by_chrom <span class="se">\</span>
        --cwd $<span class="o">[</span>_output:d<span class="o">]</span>  <span class="se">\</span>
        --phenoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]</span>:a<span class="o">]</span> <span class="se">\</span>
        --region-list $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]</span>:a<span class="o">]</span> <span class="se">\</span>
        --container $<span class="o">[</span>container_rnaquant<span class="o">]</span> <span class="se">\</span>
        --mem 4G      $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="genotype-processing">
<h3>Genotype Processing<a class="headerlink" href="#genotype-processing" title="Permalink to this headline">¶</a></h3>
<p>Since genotype is shared among the eight tissue, the QC of whole genome file is not needed. Only pca needed to be run again.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>sample_match<span class="o">]</span>
input: <span class="nv">for_each</span> <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd:a}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/{sample_participant_lookup:bn}.filtered.txt&#39;</span>,
        <span class="nv">geno</span> <span class="o">=</span> f<span class="s1">&#39;{cwd:a}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/{sample_participant_lookup:bn}.filtered_geno.txt&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/sample_matcher.ipynb filtered_sample_list <span class="se">\</span>
        --cwd $<span class="o">[</span>_output<span class="o">[</span><span class="m">0</span><span class="o">]</span>:d<span class="o">]</span>  <span class="se">\</span>
        --phenoFile $<span class="o">[</span>_input_inv<span class="o">[</span><span class="s2">&quot;molecular_pheno&quot;</span><span class="o">]]</span>  <span class="se">\</span>
        --genoFile $<span class="o">[</span>path<span class="o">(</span>_input_inv<span class="o">[</span><span class="s2">&quot;genotype_file&quot;</span><span class="o">])</span>:n<span class="o">]</span>.fam  <span class="se">\</span>
        --sample-participant-lookup $<span class="o">[</span>sample_participant_lookup<span class="o">]</span> <span class="se">\</span>
        --container $<span class="o">[</span>container_rnaquant<span class="o">]</span> <span class="se">\</span>
        --translated_phenoFile $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>king<span class="o">]</span>
input:output_from<span class="o">(</span><span class="s2">&quot;sample_match&quot;</span><span class="o">)[</span><span class="s2">&quot;geno&quot;</span><span class="o">]</span>, <span class="nv">group_with</span> <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: <span class="nv">related</span> <span class="o">=</span> f<span class="s1">&#39;{cwd:a}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/genotype_data/{path(_input_inv[&quot;genotype_file&quot;]):bn}.{_input_inv[&quot;Theme&quot;]}.related.bed&#39;</span>,
        <span class="nv">unrelated</span> <span class="o">=</span> f<span class="s1">&#39;{cwd:a}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/genotype_data/{path(_input_inv[&quot;genotype_file&quot;]):bn}.{_input_inv[&quot;Theme&quot;]}.unrelated.bed&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
 sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/GWAS_QC.ipynb king <span class="se">\</span>
    --cwd $<span class="o">[</span>_output<span class="o">[</span><span class="m">0</span><span class="o">]</span>:d<span class="o">]</span> <span class="se">\</span>
    --genoFile $<span class="o">[</span>_input_inv<span class="o">[</span><span class="s2">&quot;genotype_file&quot;</span><span class="o">]]</span> <span class="se">\</span>
    --name $<span class="o">[</span>_input_inv<span class="o">[</span><span class="s2">&quot;Theme&quot;</span><span class="o">]]</span> <span class="se">\</span>
    --keep-samples $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
    --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span> <span class="se">\</span>
    --walltime 48h  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>unrelated_QC<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;king&quot;</span><span class="o">)[</span><span class="s2">&quot;unrelated&quot;</span><span class="o">]</span>
output: <span class="nv">unrelated_bed</span> <span class="o">=</span> f<span class="s1">&#39;{_input:n}.filtered.prune.bed&#39;</span>, 
        <span class="nv">prune</span> <span class="o">=</span> f<span class="s1">&#39;{_input:n}.filtered.prune.in&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
 sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/GWAS_QC.ipynb qc <span class="se">\</span>
    --cwd $<span class="o">[</span>_output<span class="o">[</span><span class="m">0</span><span class="o">]</span>:d<span class="o">]</span> <span class="se">\</span>
    --genoFile $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
    --exclude-variants /mnt/mfs/statgen/snuc_pseudo_bulk/Ast/genotype/dupe_snp_to_exclude <span class="se">\</span>
    --maf-filter <span class="m">0</span>.05 <span class="se">\</span>
    --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span> <span class="se">\</span>
    --mem 40G  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span> 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>related_QC<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;king&quot;</span><span class="o">)[</span><span class="s2">&quot;related&quot;</span><span class="o">]</span>,output_from<span class="o">(</span><span class="s2">&quot;unrelated_QC&quot;</span><span class="o">)[</span><span class="s2">&quot;prune&quot;</span><span class="o">]</span>
output: f<span class="s1">&#39;{_input[0]:n}.filtered.extracted.bed&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
 sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/GWAS_QC.ipynb qc_no_prune <span class="se">\</span>
    --cwd $<span class="o">[</span>_output<span class="o">[</span><span class="m">0</span><span class="o">]</span>:d<span class="o">]</span> <span class="se">\</span>
    --genoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span> <span class="se">\</span>
    --exclude-variants /mnt/mfs/statgen/snuc_pseudo_bulk/Ast/genotype/dupe_snp_to_exclude <span class="se">\</span>
    --maf-filter <span class="m">0</span> <span class="se">\</span>
    --geno-filter <span class="m">0</span> <span class="se">\</span>
    --mind-filter <span class="m">0</span>.1 <span class="se">\</span>
    --hwe-filter <span class="m">0</span> <span class="se">\</span>
    --keep-variants $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span>  <span class="se">\</span>
    --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span>  <span class="se">\</span>
    --mem 40G   $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="factor-analysis">
<h2>Factor analysis<a class="headerlink" href="#factor-analysis" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>pca<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;unrelated_QC&quot;</span><span class="o">)[</span><span class="s2">&quot;unrelated_bed&quot;</span><span class="o">]</span>,group_with <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/pca/{_input:bn}.pca.rds&#39;</span>,
        f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/pca/{_input:bn}.pca.scree.txt&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
     sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/PCA.ipynb flashpca <span class="se">\</span>
        --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
        --genoFile $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
        --container $<span class="o">[</span>container_flashpca<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>projected_sample<span class="o">]</span>
<span class="c1"># The percentage of PVE explained</span>
parameter: <span class="nv">PVE_treshold</span> <span class="o">=</span> <span class="m">0</span>.7
input: output_from<span class="o">(</span><span class="s2">&quot;related_QC&quot;</span><span class="o">)</span>,output_from<span class="o">(</span><span class="s2">&quot;pca&quot;</span><span class="o">)</span>, <span class="nv">group_with</span> <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/pca/{_input[0]:bn}.pca.projected.rds&#39;</span>,
        f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/pca/{_input[0]:bn}.pca.projected.scree.txt&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/PCA.ipynb project_samples <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --genoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span> <span class="se">\</span>
            --pca-model  $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span> <span class="se">\</span>
            --maha-k <span class="sb">`</span>awk <span class="s1">&#39;$3 &lt; $[PVE_treshold]&#39;</span> $<span class="o">[</span>_input<span class="o">[</span><span class="m">2</span><span class="o">]]</span> <span class="p">|</span> tail -1 <span class="p">|</span> cut -f <span class="m">1</span>  <span class="sb">`</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_flashpca<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>merge_pca_covariate<span class="o">]</span>
<span class="c1"># The percentage of PVE explained</span>
parameter: <span class="nv">PVE_treshold</span> <span class="o">=</span> <span class="m">0</span>.7
input: output_from<span class="o">(</span><span class="s2">&quot;projected_sample&quot;</span><span class="o">)</span>,group_with <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/covariates/{path(_input_inv[&quot;covariate_file&quot;]):bn}.pca.cov.gz&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/covariate_formatting.ipynb merge_pca_covariate <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --pcaFile $<span class="o">[</span>_input:a<span class="o">]</span> <span class="se">\</span>
            --covFile  $<span class="o">[</span>path<span class="o">(</span>_input_inv<span class="o">[</span><span class="s2">&quot;covariate_file&quot;</span><span class="o">])]</span> <span class="se">\</span>
            --tol_cov <span class="m">0</span>.3  <span class="se">\</span>
            --k <span class="sb">`</span>awk <span class="s1">&#39;$3 &lt; $[PVE_treshold]&#39;</span> $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span> <span class="p">|</span> tail -1 <span class="p">|</span> cut -f <span class="m">1</span> <span class="sb">`</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>resid_exp<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;merge_pca_covariate&quot;</span><span class="o">)</span>,output_from<span class="o">(</span><span class="s2">&quot;annotation&quot;</span><span class="o">)</span>,group_with <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/phenotype/{_input[1]:bnn}.resid.bed.gz&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/covariate_formatting.ipynb Resicompute_residual <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --phenoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]</span>:a<span class="o">]</span> <span class="se">\</span>
            --covFile  $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]</span>:a<span class="o">]</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>factor<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;resid_exp&quot;</span><span class="o">)</span>,group_with <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/covariates/{_input[0]:bnn}.{factor_option}.cov.gz&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
         sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/$<span class="o">[</span>factor_option<span class="o">]</span>_factor.ipynb $<span class="o">[</span>factor_option<span class="o">]</span> <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --phenoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]</span>:a<span class="o">]</span>  <span class="se">\</span>
            --container $<span class="o">[</span>container_apex <span class="k">if</span> <span class="nv">factor_option</span> <span class="o">==</span> <span class="s2">&quot;BiCV&quot;</span> <span class="k">else</span> container_PEER<span class="o">]</span>  <span class="se">\</span>
            --walltime 24h <span class="se">\</span>
            --numThreads <span class="m">8</span> <span class="se">\</span>
            --iteration <span class="m">1000</span> <span class="se">\</span>
            --N <span class="m">60</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>merge_factor_covariate<span class="o">]</span>
<span class="c1"># The percentage of PVE explained</span>
parameter: <span class="nv">PVE_treshold</span> <span class="o">=</span> <span class="m">0</span>.7
input: output_from<span class="o">(</span><span class="s2">&quot;factor&quot;</span><span class="o">)</span>,output_from<span class="o">(</span><span class="s2">&quot;merge_pca_covariate&quot;</span><span class="o">)</span>,group_with <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd}/data_preprocessing/{_input_inv[&quot;Theme&quot;]}/covariates/{path(_input_inv[&quot;covariate_file&quot;]):bn}.pca.{factor_option}.cov.gz&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/covariate_formatting.ipynb merge_factor_covariate <span class="se">\</span>
            --cwd $<span class="o">[</span>_output:d<span class="o">]</span> <span class="se">\</span>
            --factorFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]</span>:a<span class="o">]</span> <span class="se">\</span>
            --covFile  $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]</span>:a<span class="o">]</span> <span class="se">\</span>
            --container $<span class="o">[</span>container_base_bioinfo<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="association-scan">
<h2>Association scan<a class="headerlink" href="#association-scan" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>TensorQTL<span class="o">]</span>
parameter: <span class="nv">genotype_list</span> <span class="o">=</span> path
input: genotype_list, output_from<span class="o">(</span><span class="s2">&quot;phenotype_partition_by_chrom&quot;</span><span class="o">)</span>,output_from<span class="o">(</span><span class="s2">&quot;merge_factor_covariate&quot;</span><span class="o">)</span>,group_with <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd:a}/Association_scan/{_input_inv[&quot;Theme&quot;]}/TensorQTL/TensorQTL_recipe.tsv&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/TensorQTL.ipynb TensorQTL_cis <span class="se">\</span>
        --genotype_file_list $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span>   <span class="se">\</span>
        --molecular_pheno_list $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span> <span class="se">\</span>
        --covariate $<span class="o">[</span>_input<span class="o">[</span><span class="m">2</span><span class="o">]]</span>    <span class="se">\</span>
        --cwd $<span class="o">[</span>_output:d<span class="o">]</span>  <span class="se">\</span>
        --container $<span class="o">[</span>container_TensorQTL<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>APEX<span class="o">]</span>
parameter: <span class="nv">genotype_list</span> <span class="o">=</span> path
input: output_from<span class="o">(</span><span class="s2">&quot;plink_to_vcf&quot;</span><span class="o">)</span>, output_from<span class="o">(</span><span class="s2">&quot;phenotype_partition_by_chrom&quot;</span><span class="o">)</span>,output_from<span class="o">(</span><span class="s2">&quot;merge_factor_covariate&quot;</span><span class="o">)</span>,group_with <span class="o">=</span> <span class="s2">&quot;input_inv&quot;</span>
output: f<span class="s1">&#39;{cwd:a}/Association_scan/{_input_inv[&quot;Theme&quot;]}/APEX/APEX_recipe.tsv&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>, <span class="nv">stderr</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stderr&#39;</span>, <span class="nv">stdout</span> <span class="o">=</span> f<span class="s1">&#39;{_output[0]}.stdout&#39;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/APEX.ipynb APEX_cis <span class="se">\</span>
        --genotype_file_list $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span>   <span class="se">\</span>
        --molecular_pheno_list $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span> <span class="se">\</span>
        --covariate $<span class="o">[</span>_input<span class="o">[</span><span class="m">2</span><span class="o">]]</span>    <span class="se">\</span>
        --cwd $<span class="o">[</span>_output:d<span class="o">]</span>  <span class="se">\</span>
        --container $<span class="o">[</span>container_apex<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="susie">
<h2>SuSiE<a class="headerlink" href="#susie" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>UniSuSiE<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;plink_per_gene&quot;</span><span class="o">)</span>, output_from<span class="o">(</span><span class="s2">&quot;annotation&quot;</span><span class="o">)</span>,output_from<span class="o">(</span><span class="s2">&quot;factor&quot;</span><span class="o">)</span>, output_from<span class="o">(</span><span class="s2">&quot;region_list_concat&quot;</span><span class="o">)</span>, <span class="nv">group_by</span> <span class="o">=</span> <span class="s2">&quot;all&quot;</span>
output: f<span class="s1">&#39;{cwd:a}/Fine_mapping/UniSuSiE/UniSuSiE_recipe.tsv&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>
    sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/SuSiE.ipynb uni_susie <span class="se">\</span>
        --genoFile $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span>   <span class="se">\</span>
        --phenoFile $<span class="o">[</span><span class="s2">&quot; &quot;</span>.join<span class="o">([</span>str<span class="o">(</span>x<span class="o">)</span> <span class="k">for</span> x <span class="k">in</span> _input<span class="o">[</span><span class="m">1</span>:len<span class="o">(</span>input_inv<span class="o">)</span>+1<span class="o">]])]</span> <span class="se">\</span>
        --covFile $<span class="o">[</span><span class="s2">&quot; &quot;</span>.join<span class="o">([</span>str<span class="o">(</span>x<span class="o">)</span> <span class="k">for</span> x <span class="k">in</span>  _input<span class="o">[</span>len<span class="o">(</span>input_inv<span class="o">)</span>+1:len<span class="o">(</span>input_inv<span class="o">)</span>*2+1<span class="o">]])]</span>  <span class="se">\</span>
        --cwd $<span class="o">[</span>_output:d<span class="o">]</span>  <span class="se">\</span>
        --tissues $<span class="o">[</span><span class="s2">&quot; &quot;</span>.join<span class="o">([</span>x<span class="o">[</span><span class="s2">&quot;Theme&quot;</span><span class="o">]</span> <span class="k">for</span> x <span class="k">in</span> input_inv<span class="o">])]</span> <span class="se">\</span>
        --region-list $<span class="o">[</span>_input<span class="o">[</span><span class="m">3</span><span class="o">]]</span> <span class="se">\</span>
        --container $<span class="o">[</span>container_susie<span class="o">]</span> $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="sumstat-merger">
<h2>Sumstat Merger<a class="headerlink" href="#sumstat-merger" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>yml_generation<span class="o">]</span>
input: output_from<span class="o">(</span>Association_option<span class="o">)</span>
output: f<span class="s1">&#39;{cwd:a}/Data_intergration/qced_sumstat_list.txt&#39;</span>,f<span class="s1">&#39;{cwd:a}/Data_intergration/{Association_option}/yml_list.txt&#39;</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>
        sos run $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/yml_generator.ipynb yml_list <span class="se">\</span>
            --sumstat-list $<span class="o">[</span>_input<span class="o">]</span> <span class="se">\</span>
            --cwd  $<span class="o">[</span>_output<span class="o">[</span><span class="m">0</span><span class="o">]</span>:d<span class="o">]</span> --name $<span class="o">[</span><span class="s2">&quot; &quot;</span>.join<span class="o">([</span>str<span class="o">(</span>x<span class="o">)</span>.split<span class="o">(</span><span class="s2">&quot;/&quot;</span><span class="o">)[</span>-3<span class="o">]</span> <span class="k">for</span> x <span class="k">in</span> _input<span class="o">])]</span> 

         
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="o">[</span>sumstat_merge<span class="o">]</span>
input: output_from<span class="o">(</span><span class="s2">&quot;yml_generation&quot;</span><span class="o">)</span>
report:  <span class="nv">expand</span> <span class="o">=</span> <span class="s2">&quot;</span>$<span class="s2">[ ]&quot;</span>
      sos run  $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/summary_stats_merger.ipynb     <span class="se">\</span>
            --sumstat-list $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span>    <span class="se">\</span>
            --yml-list $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span>    <span class="se">\</span>
            --cwd $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]</span>:d<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span> --mem 50G --walltime 48h 
          
      sos run  $<span class="o">[</span>exe_dir<span class="o">]</span>/pipeline/summary_stats_merger.ipynb    sumstat_to_vcf  <span class="se">\</span>
         --sumstat-list $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]]</span>    <span class="se">\</span>
         --yml-list $<span class="o">[</span>_input<span class="o">[</span><span class="m">1</span><span class="o">]]</span>    <span class="se">\</span>
         --cwd $<span class="o">[</span>_input<span class="o">[</span><span class="m">0</span><span class="o">]</span>:d<span class="o">]</span>  $<span class="o">[</span>submission <span class="k">if</span> yml.is_file<span class="o">()</span> <span class="k">else</span> <span class="s2">&quot;&quot;</span> <span class="o">]</span> --mem 50G --walltime 48h 
      
</pre></div>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "bash"
        },
        kernelOptions: {
            kernelName: "bash",
            path: "./code/commands_generator"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'bash'</script>

              </div>
              
        
            <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="bulk_expression_commands.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">RNA-seq calling and QC</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../data_preprocessing/reference_data.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Reference data standardization</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
        
        </div>
    </div>
    <footer class="footer">
    <div class="container">
      <p>
        
          By The NIH/NIA Alzheimer's Disease Sequencing Project Functional Genomics Consortium<br/>
        
            &copy; Copyright 2021, ADSP-FG xQTL methods and data integration subgroup.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>